blockchain/tx_pool.go

```diff
@@ -658,38 +668,42 @@ func (pool *TxPool) validateTx(tx *types.Transaction) error {
 			return ErrFeeCapVeryHigh
 		}
 
-		// TODO : Need to KIP-71 hardfork.
-		// Ensure gasFeeCap is greater than or equal to gasTipCap.
-		//if tx.GasFeeCap().Cmp(tx.GasTipCap()) < 0 {
-		//	return ErrTipAboveFeeCap
-		//}
-		//
-		//if pool.gasPrice.Cmp(tx.GasTipCap()) != 0 {
-		//	logger.Trace("fail to validate maxPriorityFeePerGas", "unitprice", pool.gasPrice, "maxPriorityFeePerGas", tx.GasFeeCap())
-		//	return ErrInvalidGasTipCap
-		//}
-		//
-		//if pool.gasPrice.Cmp(tx.GasFeeCap()) != 0 {
-		//	logger.Trace("fail to validate maxFeePerGas", "unitprice", pool.gasPrice, "maxFeePerGas", tx.GasTipCap())
-		//	return ErrInvalidGasFeeCap
-		//}
-
-		// Ensure transaction's gasFeeCap is greater than or equal to transaction pool's gasPrice(baseFee).
-		if pool.gasPrice.Cmp(tx.GasFeeCap()) > 0 {
-			logger.Trace("fail to validate maxFeePerGas", "unitPrice", pool.gasPrice, "maxFeePerGas", tx.GasFeeCap())
-			return ErrFeeCapBelowBaseFee
+		if pool.kip71 {
+			// Ensure transaction's gasFeeCap is greater than or equal to transaction pool's gasPrice(baseFee).
+			if pool.gasPrice.Cmp(tx.GasFeeCap()) > 0 {
+				logger.Trace("fail to validate maxFeePerGas", "pool.gasPrice", pool.gasPrice, "maxFeePerGas", tx.GasFeeCap())
+				return ErrFeeCapBelowBaseFee
+			}
+		} else {
+			// Ensure gasFeeCap is greater than or equal to gasTipCap.
+			if tx.GasFeeCap().Cmp(tx.GasTipCap()) < 0 {
```

Comment:
I think we can make this condition as a common condition (not only when `!pool.kip71`).

Reply:
You mean It's better to make a function to check this condition?

Reply:
I thought we will keep this condition after hard fork also.. maybe i misunderstood?